assets/aws/files/bin/teleport-generate-config

```diff
@@ -3,92 +3,204 @@ if [[ "${DEBUG}" == "true" ]]; then
+        else
+            KUBERNETES_PUBLIC_ADDR="'${TELEPORT_PROXY_SERVER_LB}:3026'"
+        fi
     else
-      KUBERNETES_PUBLIC_ADDR="'${TELEPORT_PROXY_SERVER_LB}:3026'"
+        KUBERNETES_PUBLIC_ADDR="'${EXTERNAL_HOSTNAME:-$PUBLIC_IP}:3026'"
     fi
-  else
-    KUBERNETES_PUBLIC_ADDR="'${EXTERNAL_HOSTNAME:-$PUBLIC_IP}:3026'"
-  fi
-  # set up the kubernetes listener
-  cat << EOS >> /etc/teleport.yaml
+    # set up the kubernetes listener
+    cat << EOS >> ${USE_CONFIG_PATH}
   kubernetes:
     enabled: yes
     listen_addr: 0.0.0.0:3026
     public_addr: [${KUBERNETES_PUBLIC_ADDR}]
 EOS
 }
 
-write_https_keypairs_section() {
-  cat << EOS >> /etc/teleport.yaml
-  https_keypairs:
-    - cert_file: /var/lib/teleport/fullchain.pem
-      key_file: /var/lib/teleport/privkey.pem
+write_ssh_and_tunnel_section() {
+    PASSED_EXTERNAL_TUNNEL_PORT="$1"
+    SSH_PORT=3023
+    if [[ "${TELEPORT_PROXY_SERVER_LB}" != "" ]]; then
+        # ACM
+        if [[ "${USE_ACM}" == "true" ]]; then
+            # if an alias for the ACM NLB is configured, use that for the proxy's ssh_public_addr
+            if [[ "${TELEPORT_PROXY_SERVER_NLB_ALIAS}" != "" ]]; then
+                SSH_HOSTNAME="${TELEPORT_PROXY_SERVER_NLB_ALIAS}"
+            else
+                SSH_HOSTNAME="${TELEPORT_PROXY_SERVER_LB}"
+            fi
+        # non-ACM
+        else
+            if [[ "${TELEPORT_DOMAIN_NAME}" != "" ]]; then
+                SSH_HOSTNAME="${TELEPORT_DOMAIN_NAME}"
+            else
+                SSH_HOSTNAME="${TELEPORT_EXTERNAL_HOSTNAME:-$PUBLIC_IP}"
+            fi
+        fi
+        # when using ACM, we have to set up a dedicated listener and target group as
+        # SSL termination is handled by the ACM ALB and it can't multiplex the tunnel connections
+        if [[ "${USE_ACM}" == "true" ]]; then
+            TUNNEL_PORT="${TUNNEL_LISTEN_PORT}"
+            if [[ "${TELEPORT_PROXY_SERVER_NLB_ALIAS}" != "" ]]; then
+                TUNNEL_HOSTNAME="${TELEPORT_PROXY_SERVER_NLB_ALIAS}"
+            else
+                TUNNEL_HOSTNAME="${TELEPORT_PROXY_SERVER_LB}"
+            fi
+        # when ACM is not being used, Teleport handles SSL termination itself and so tunnel connections
+        # can be multiplexed over the same connection as regular web proxy traffic
+        # this is automatically enabled when Teleport's web and tunnel listeners are configured to use the same port
+        else
+            TUNNEL_PORT="${PASSED_EXTERNAL_TUNNEL_PORT}"
+            # check whether an external domain name is configured (HA mode)
+            # if not (starter mode), use an external hostname if configured, and fall back to public IP
+            if [[ "${TELEPORT_DOMAIN_NAME}" != "" ]]; then
+                TUNNEL_HOSTNAME="${TELEPORT_DOMAIN_NAME}"
+            else
+                TUNNEL_HOSTNAME="${TELEPORT_EXTERNAL_HOSTNAME:-$PUBLIC_IP}"
+            fi
+        fi
```

Comment:
nit: can these conditional blocks be merged?
they do the same checks and only differ in what vars they set